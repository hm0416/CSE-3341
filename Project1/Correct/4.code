
program int X , Y , XY 	;

          input X, Y, XY;
		  
		  if !(X = Y)
		  then
			if X < Y
			then
				output 1;
			else
				output 2;
			endif;
		output 3;
		endif;
		  
		input X, Y, XY;
		  
		  if !(X = Y)
		  then
			if X < Y
			then
				output 10;
			else
				output 20;
			endif;
		output 30;
		endif;
		
		input X, Y, XY;
		  
		  if !(X = Y)
		  then
			if X < Y
			then
				output 100;
			else
				output 200;
			endif;
		output 300;
		endif;

		input X, Y, XY;
		  
		if X = Y 
		then
			output 199;
		else
			if !(X <= Y)
			then
				output 299;
			else
				output 399;
			endif;
		endif;
		
		
          end  
      
