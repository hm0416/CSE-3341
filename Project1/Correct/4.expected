PROGRAM
INT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
INPUT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
NEGATION
LPAREN
ID[X]
ASSIGN
ID[Y]
RPAREN
THEN
IF
ID[X]
LESS
ID[Y]
THEN
OUTPUT
CONST[1]
SEMICOLON
ELSE
OUTPUT
CONST[2]
SEMICOLON
ENDIF
SEMICOLON
OUTPUT
CONST[3]
SEMICOLON
ENDIF
SEMICOLON
INPUT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
NEGATION
LPAREN
ID[X]
ASSIGN
ID[Y]
RPAREN
THEN
IF
ID[X]
LESS
ID[Y]
THEN
OUTPUT
CONST[10]
SEMICOLON
ELSE
OUTPUT
CONST[20]
SEMICOLON
ENDIF
SEMICOLON
OUTPUT
CONST[30]
SEMICOLON
ENDIF
SEMICOLON
INPUT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
NEGATION
LPAREN
ID[X]
ASSIGN
ID[Y]
RPAREN
THEN
IF
ID[X]
LESS
ID[Y]
THEN
OUTPUT
CONST[100]
SEMICOLON
ELSE
OUTPUT
CONST[200]
SEMICOLON
ENDIF
SEMICOLON
OUTPUT
CONST[300]
SEMICOLON
ENDIF
SEMICOLON
INPUT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IF
ID[X]
ASSIGN
ID[Y]
THEN
OUTPUT
CONST[199]
SEMICOLON
ELSE
IF
NEGATION
LPAREN
ID[X]
LESSEQUAL
ID[Y]
RPAREN
THEN
OUTPUT
CONST[299]
SEMICOLON
ELSE
OUTPUT
CONST[399]
SEMICOLON
ENDIF
SEMICOLON
ENDIF
SEMICOLON
END
